### PR TITLE
change pre-commit config .yaml file

### PR DESCRIPTION
**Pull request by zhen0427**

### CODE REVIEW

**Comment by mgovers** [comment]
please tell me the windows runners are not broken again

```
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.40.33807\include\yvals_core.h(898,1): error: static assertion failed: error STL1000: Unexpected compiler version, expected Clang 17.0.0 or newer.
_EMIT_STL_ERROR(STL1000, "Unexpected compiler version, expected Clang 17.0.0 or newer.");
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.40.33807\include\yvals_core.h(519,44): note: expanded from macro '_EMIT_STL_ERROR'
#define _EMIT_STL_ERROR(NUMBER, MESSAGE)   static_assert(false, "error " #NUMBER ": " MESSAGE)
```

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -4,19 +4,19 @@
 
 repos:
   - repo: https://github.com/fsfe/reuse-tool
-    rev: v3.0.1
+    rev: v3.1.0a1
```
Comment (mgovers) [inline .pre-commit-config.yaml:7]:
```suggestion
    rev: v3.0.2
```

**Comment by mgovers** [comment]
build fails due to https://github.com/actions/runner-images/issues/10001

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Jerry-Jinfeng-Guo) [reply .pre-commit-config.yaml:7]: So?

**Review by Jerry-Jinfeng-Guo — APPROVED** [review]

**Pull request merged**